text/0030-asc-desc-criterion.md

```diff
@@ -0,0 +1,91 @@
+- MeiliSearch Tracking-Issues: [#161](https://github.com/meilisearch/milli/issues/161)
+
+## 1. Feature Description and Interaction
+
+### I. Summary
+
+Ranking rules are built-in rules that ensure relevancy in search results. Ranking rules are applied in a default order which can be changed in the settings. You can add or remove rules and change their order of importance.
+
+MeiliSearch allows you to create custom rules within the default rules. Custom rules are dedicated to sorting in ascending or descending order on an attribute.
+
+### II. Motivation
+
+We want to provide our users with an always improved usage experience. Relevance is essential in a search engine since it is what allows the engine to fulfill its objective. Delivering results that match user demands by allowing modification of the relevance is critical.
+
+The new search engine called Milli no longer processes this criterion of relevance as the current MeilliSearch. This specification makes it possible to state that Milli will be identical to the ISO version (v0.20) in the usage of API interfaces but also in the expected search results.
+
+### III. Additional Materials
+
+#### Algolia
+
+Algolia offers 8 classification rules to achieve relevance.
+
+- Number of typos
+- Geolocation
+- Number of words in the query matching in the result
+- Filters
+- Distance between words
+- Best matching attribute in the record
+- Number of words matching exactly (without typo)
+- Custom ranking
+
+Note that Algolia don't recommend to change the order of the default criteria because of the fact that it works for the vast majority of their use cases. Like MeiliSearch (v0.20), document that don’t have an attribute that’s in the customRanking list are pushed to the bottom of the search result.
+
+### IV.Explanation
+
+#### Current behavior of v0.20
+
+In the case of certain documents that do not contain the attribute configured on a custom ASC/DESC ranking rules, it will return those documents at the bottom of the search result. Documents are not excluded and can be returned in the search result.
+
+> In the case of several custom ASC / DESC rules configured within the rankings rules, the sorting behavior is difficult to explain and can appear undefined when the documents do not necessarily contain the attributes which are set on this rule.
+
+This criterion can only be used with an attribute containing numbers. 
+
+#### Current behavior of Milli
+
+If the attribute on which the ASC / DESC criterion is configured does not exist in all the documents of the index concerned by the search, Milli will discard them and never return them. This behavior is not necessarily what a end user would expect when searching.
+
+As v0.20, Milli can’t handle ASC / DESC criterion on string. Only numbers allow its use.
+
+#### Decisions
+
+1. We thought about allowing the use of the ASC / DESC criterion in the ranking rules settings as long as it would also be declared in the attributesForFaceting.
```

Comment:
I'm not sure this is what I remember. 
Was it not: 
- You can add any attribute as `ASC / DESC` custom ranking rule, but they will not be considered filterable. 
- In order for an attribute that is present in `ASC / DESC ` custom ranking rule to be filterable, it has to be added in `attributesForFaceting` as well. 

Maybe I'm wrong :s

Reply:
Yes, these are the final decisions. We were wondering if it needed to be declared in attributesForFaceting to be set as a custom ranking rule. The fact that it's not filterable if the field is not set in attributesForFaceting is for clarity and convenience in the API usage.

Reply:
Ah oke! I see 
```suggestion
1. We thought about allowing filtering of the attributes used in the ASC / DESC as long as it would also be declared in the attributesForFaceting.
```

This seems to be more clear but I would rephrase it like that.
```suggestion
1. Attributes used in ASC/DESC criterion doesn't make them filterable during search. They should be added in `attributesForFaceting` to filter on it.
```

What do you think?

Reply:
I am going to add the first proposition because it shows that we have thought about this subject. In addition, it follows the convention of the second question mark just below. Thanks @bidoubiwa!